examples/utils/utils.go

```diff
@@ -652,3 +660,20 @@ func clientSetFromKubeconfig(kubeconfig any) (*kubernetes.Clientset, error) {
 	}
 	return clientSet, nil
 }
+
+// EnsureKubeconfigFails ensures that the provided kubeconfig fails to authenticate.
+func EnsureKubeconfigFails(t *testing.T, kubeconfig any) {
+	kc, err := json.Marshal(kubeconfig)
+	if err != nil {
+		t.Errorf("unable to marshal provided kubeconfig: %s", err)
+	}
+	kubeAccess, err := KubeconfigToKubeAccess(kc)
+	if err != nil {
+		t.Errorf("unable to create KubeAccess from kubeconfig: %s", err)
+	}
+
+	_, err = kubeAccess.Clientset.Discovery().ServerVersion()
```

Comment:
It would be good to distinguish auth errors from other errors so we don't hide those.

Reply:
I'm working on another PR to improve our e2e tests, so I'll roll this suggestion into that PR as well!